flink-connectors/flink-connector-kafka-0.9/src/test/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThreadTest.java

```diff
@@ -744,17 +745,21 @@ void reassignPartitions(List<KafkaTopicPartitionState<TopicPartition>> newPartit
 			final OneShotLatch continueAssignmentLatch) {
 
 		final KafkaConsumer<byte[], byte[]> mockConsumer = mock(KafkaConsumer.class);
+		final AtomicInteger callCounter = new AtomicInteger();
+
 		when(mockConsumer.assignment()).thenAnswer(new Answer<Object>() {
 			@Override
 			public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
-				if (midAssignmentLatch != null) {
-					midAssignmentLatch.trigger();
-				}
+				// first call is not the one that we want to catch... we all love mocks, don't we?
```

Comment:
Could you explain a bit on "first call is not the one that we want to catch"? Which test was failing? I have the feeling that this could have been fixed in a different way.